hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/PollMCEventsMessageTask.java

```diff
@@ -45,7 +45,7 @@ protected Object call() throws Exception {
         if (mcs == null) {
             return Collections.<MCEventDTO>emptyList();
         }
-        List<Event> polledEvents = mcs.pollMCEvents();
+        List<Event> polledEvents = mcs.pollMCEvents(connection.getRemoteAddress());
```

Comment:
There is a potential issue related with `connection.getRemoteAddress()` which I forgot about during the initial review. Namely, it's not unique since it uses `java.net.Socket#getRemoteSocketAddress()` under the hood. If multiple clients (MC instances) are located behind a NAT, all of them will get the same address as the result of this call. We probably should use client UUID instead to identify MC instances reliably. WDYT?

Reply:
Makes sense, thanks for raising this, done.